clang/test/CodeGen/cheri/cheri-asm-ptrtoint.c

```diff
@@ -0,0 +1,31 @@
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --version 2
+// RUN: %riscv32_cheri_purecap_cc1 -S -emit-llvm -O1 -o - %s -verify | FileCheck %s --check-prefix=RV32
+// RUN: %riscv64_cheri_purecap_cc1 -S -emit-llvm -O1 -o - %s -verify | FileCheck %s --check-prefix=RV64
+// RUN: not %riscv32_cheri_purecap_cc1 -S -o - %s 2>&1 | FileCheck %s --check-prefix=ERR
+// RUN: not %riscv64_cheri_purecap_cc1 -S -o - %s 2>&1 | FileCheck %s --check-prefix=ERR
+//
+// This asm syntax will fail when it gets to the backend, but at least prevent
+// a crash in clang and display the correct error message.
```

Comment:
Please don't introduce new tests that use the backend. If you really want to test the error message, add a test in llvm/test/CodeGen, but I don't think it's all that interesting.